src/main/scala/uk/gov/nationalarchives/omega/api/business/records/GetRecordService.scala

```diff
@@ -0,0 +1,328 @@
+      recordProperties.groupBy(_.recordDescriptionUri.toString)
+    val recordDescriptionProperties: List[RecordDescriptionPropertiesEntity] =
+      recordPropertiesMap.getOrElse(descriptionUri, List.empty)
+    recordDescriptionProperties.headOption match {
+      case Some(props) => propertiesFromEntity(props, isReferencedByMap, relatedToMap, separatedFromMap, subjectMap)
+      case None        => RecordDescriptionProperties()
+    }
+  }
+
+  private def propertiesFromEntity(
+    entity: RecordDescriptionPropertiesEntity,
+    isReferencedByMap: Map[String, List[LabelledIdentifier]],
+    relatedToMap: Map[String, List[LabelledIdentifier]],
+    separatedFromMap: Map[String, List[LabelledIdentifier]],
+    subjectMap: Map[String, List[Identifier]]
+  ): RecordDescriptionProperties =
+    RecordDescriptionProperties(
+      assetLegalStatus = getLegalStatus(entity),
+      legacyTnaCs13RecordType = getCS13RecordType(entity),
+      designationOfEdition = entity.designationOfEdition,
+      created = getCreated(entity),
+      archivistsNote = entity.archivistsNote,
+      sourceOfAcquisition = entity.sourceOfAcquisition.flatMap(uri => Some(GeneralIdentifier(uri))),
+      custodialHistory = entity.custodialHistory,
+      administrativeBiographicalBackground = entity.adminBiogBackground,
+      accumulation = getAccumulation(entity),
+      appraisal = entity.appraisal,
+      accrualPolicy = entity.accrualPolicy.flatMap(uri => Some(GeneralIdentifier(uri))),
+      layout = entity.layout,
+      publicationNote = entity.publicationNote,
+      isReferencedBy = isReferencedByMap.get(entity.recordDescriptionUri),
+      relatedTo = relatedToMap.get(entity.recordDescriptionUri),
+      separatedFrom = separatedFromMap.get(entity.recordDescriptionUri),
+      subject = subjectMap.get(entity.recordDescriptionUri)
+    )
+
+  private def getLegalStatus(entity: RecordDescriptionPropertiesEntity): Option[LabelledIdentifier] =
+    for {
+      legalStatusUri   <- entity.assetLegalStatus
+      legalStatusLabel <- entity.legalStatusLabel
+    } yield LabelledIdentifier(legalStatusUri, legalStatusLabel)
+
+  private def getCS13RecordType(entity: RecordDescriptionPropertiesEntity): Option[CS13RecordType] =
+    entity.legacyType.flatMap { uri =>
+      CS13RecordType.fromUri(uri) match {
+        case Success(recordType) => Some(recordType)
+        case _                   => None // TODO (RW) log error here (see PACT-1071)
+      }
+    }
+
+  private def getCreated(entity: RecordDescriptionPropertiesEntity): Option[DescribedTemporal] =
+    entity.createdType.flatMap(uri =>
+      uri.toString match {
+        case s"${BaseURL.time}ProperInterval" =>
+          for {
+            description <- entity.createdDescription
+            dateFrom    <- entity.createdBeginning
+            dateTo      <- entity.createdEnd
+          } yield DescribedTemporal(description, TemporalInterval(dateFrom, dateTo))
+        case s"${BaseURL.time}Instant" =>
```

Comment:
As `Instant` will be used in multiple data classes, perhaps it should be a constant?
Same goes for other dataset properties...

Reply:
Now a constant in the `Time` vocabulary class.